cmd/pilot-agent/main.go

```diff
@@ -99,12 +83,46 @@ var (
 
 			glog.V(2).Infof("proxy role: %#v", role)
 
-			envoyProxy := envoy.MakeProxy(mesh, servicecluster, role.ServiceNode(), configpath, binarypath)
-			agent := proxy.NewAgent(envoyProxy, proxy.DefaultRetry)
-			watcher, err := envoy.NewWatcher(mesh, agent, role)
-			if err != nil {
+			proxyConfig := proxy.DefaultProxyConfig()
+			if config != "" {
+				if err := model.ApplyYAML(config, &proxyConfig); err != nil {
```

Comment:
Do you mean the yaml file will be passed as a string in the config ???
That doesn't make any sense - it'll be quite bad on a VM ( there are also line limits on command, etc )...

Reply:
The problem is that admission controller needs to configure the agent. We could define 20 flags and then set the values for each in the pod template. But that's error prone. This suggests to pass a proto inline in YAML form, so we can  be backwards compatible, and have stronger validation. It would look like: `./agent --config "discoveryAddress: pilot:8080"`